provide.tests/NatsClientTests.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Threading.Tasks;
+using NATS.Client;
+using Xunit;
+
+namespace provide.tests
+{
+    public class NatsClientTests
+    {
+        private async Task<NatsClient> SetupNatsClient()
+        {
+            var token = await IdentTestUtil.CreateIdentForTestUser();
+            return new NatsClient(token);
+        }
+
+        [Fact]
+        public async void TestConnection()
+        {
+            var client = await SetupNatsClient();
+            client.Connect();
+
+            Assert.Equal(ConnState.CONNECTED, client.GetConnectionState());
+        }
+
+        [Fact]
+        public async void TestSubscribeAndUnsubscribe() 
+        {
+            var client = await SetupNatsClient();
+            client.Connect();
+
+            Assert.Equal(0, client.GetSubscriptionCount());
+            client.Subscribe("network.*.status", (sender, args) => {});
```

Comment:
can not test publish at this point because publish permissions in nats prop of JWT is empty when we create using Ident, some suggestion for this?

Reply:
leave a todo for now if @kthomas does  not have some ideas

Reply:
You can vend and sign a JWT for this using the pattern illustrated here:

```
export const vendNatsAuthorization = async (natsConfig, subject): Promise<string> => {
  const authService = new AuthService(
    log,
    natsConfig?.audience || natsConfig.natsServers[0],
    natsConfig?.privateKey,
    natsConfig?.publicKey,
  );

  const permissions = {
    publish: {
      allow: ['my.pub.subject'],
    },
    subscribe: {
      allow: [`my.subscribe.subject`],
    },
  };

  return await authService.vendBearerJWT(
    subject,
    5000,
    permissions,
  );
};
```

Maybe we add a `user.<user_id>` subject by default to the NATS authorization so you can publish to your own subject, so to speak. If we added this to Ident you could just test publishing on that subject without having to implement the above.